packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts

```diff
@@ -1,219 +1,396 @@
+        }
+    }
+
+    private setCurrentTd(td: null): void;
+    private setCurrentTd(
+        td: HTMLTableCellElement,
+        tableRect: Rect,
+        right: number,
+        bottom: number
+    ): void;
+    private setCurrentTd(
+        td: HTMLTableCellElement,
+        tableRect?: Rect,
+        right?: number,
+        bottom?: number
+    ) {
+        if (this.currentTd != td) {
+            if (this.currentTd) {
+                this.resizerContainer.removeChild(this.horizontalResizer);
+                this.resizerContainer.removeChild(this.verticalResizer);
+                this.horizontalResizer = null;
+                this.verticalResizer = null;
+            }
+
+            this.currentTd = td;
+
+            if (this.currentTd) {
+                this.horizontalResizer = this.createResizer(
+                    true /*horizontal*/,
+                    tableRect.left,
+                    bottom - CELL_RESIZER_WIDTH + 1,
+                    tableRect.right - tableRect.left,
+                    CELL_RESIZER_WIDTH
+                );
+                this.verticalResizer = this.createResizer(
+                    false /*horizontal*/,
+                    right - CELL_RESIZER_WIDTH + 1,
+                    tableRect.top,
+                    CELL_RESIZER_WIDTH,
+                    tableRect.bottom - tableRect.top
+                );
+
+                this.resizerContainer.appendChild(this.horizontalResizer);
+                this.resizerContainer.appendChild(this.verticalResizer);
+            }
+        }
+    }
+
+    private createResizer(
+        horizontal: boolean,
+        left: number,
+        top: number,
+        width: number,
+        height: number
+    ) {
+        const div = fromHtml(
+            horizontal ? HORIZONTAL_RESIZER_HTML : VERTICAL_RESIZER_HTML,
+            this.editor.getDocument()
+        )[0] as HTMLDivElement;
+        div.style.top = `${top}px`;
+        div.style.left = `${left}px`;
+        div.style.width = `${width}px`;
+        div.style.height = `${height}px`;
+
+        div.addEventListener(
+            'mousedown',
+            horizontal ? this.startHorizontalResizeTable : this.startVerticalResizeTable
+        );
+
+        return div;
+    }
+
+    private startHorizontalResizeTable = (e: MouseEvent) => {
+        this.resizingState = ResizeState.Horizontal;
+        this.startResizeTable(e);
+    };
+
+    private startVerticalResizeTable = (e: MouseEvent) => {
+        this.resizingState = ResizeState.Vertical;
+        this.startResizeTable(e);
+    };
+
+    private startResizeTable(e: MouseEvent) {
+        const doc = this.editor.getDocument();
+        doc.addEventListener('mousemove', this.frameAnimateResizeTable, true);
+        doc.addEventListener('mouseup', this.endResizeTable, true);
+    }
+
+    private frameAnimateResizeTable = (e: MouseEvent) => {
+        this.resizeTableAnimationFrameID = requestAnimationFrame(() => this.resizeTable(e));

```

Comment:
What if we just handle the event directly? Since user is dragging, no other event can be triggered by user, so I think it should be fine not to run this in async way

Reply:
Just took a quick look and generally seems fine to me. I'll review into more details tomorrow.

Reply:
Okay, let me know if there is any other action that you think I need to take.I previously had throttling set up to deal with the performance of the dragging, but Will suggested using requestAnimationFrame, which made sense because this is a resizing animation.

Reply:
To run with requestAnimationFrame(), suggest call editor.runAsync() instead. It is a wrapper of requestAnimationFrame and it also do null check for editor since in theory editor may be disposed after any async call.